File: src/Modules/Hermes.Modules.Books/Commands/Books/CreateBook/CreateBookCommand.cs
using Hermes.Domain.Enums;
using Hermes.Modules.Shared.Response;
using MediatR;

namespace Hermes.Modules.Books.Commands.Books.CreateBook
{
    public class CreateBookCommand : IRequest<Response>
    {
        public string Name { get; set; } = string.Empty;
        public int PublishYear { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public bool IsAvailable { get; set; }
        public byte[]? BookImage { get; set; }
        public int AuthorId { get; set; }
        public string AuthorName { get; set; } = string.Empty;
        public string AuthorSurname { get; set; } = string.Empty;
        public int PublisherId { get; set; }
        public BookCategory Category { get; set; } = BookCategory.None;
        public int AntiqueShopId { get; set; } = 1;
    }
}

File: src/Hermes.Web/src/types/books/CreateBookCommand.ts
import BookCategory from "./BookCategory";

export default interface CreateBookCommand {
    name: string;
    publishYear: number;
    price: number;
    quantity: number;
    isAvailable: boolean;
    authorId: number;
    authorName: string;
    authorSurname: string;
    publisherId: number;
    publisherName: string;
    category: BookCategory;
    antiqueShopId: number;
    bookImage: string  

}